### ISSUE TITLE
cantools decode 200million message

### ISSUE THREAD

**Issue by Lit-fish-auto** [cantools/cantools#564]
CAN load  the asc or blf ,  about 200million message.

cantools decode these messages , and put the time ,signal name,physical value in dataframe.

overall, how to short the process time?

the way of pypy ,  multiprocessing is usefull ?

the numba do not support pandas and cantools ?

**Comment by andlaus** [comment]
> overall, how to short the process time?

for such a large files, I would recommend biting the bullet and writing your program in C or C++: for BLF there is a C++ parser (http://tomilo.org/vector_blf.html) and using `cantools genererate_c_source`, you can generate the code to decode the frames...

**Comment by Lit-fish-auto** [comment]
@andlus
can use vector_dbc  decode  data?
cantools generate c resource  can decode data  ?

**Comment by andlaus** [comment]
> can use vector_dbc decode data?

no, but it can parse your BLF file, i.e. you get a sequence of frame IDs and payload data.

> cantools generate c resource can decode data ?

in conjunction with your favourite C compiler you can use it to decode above sequence of can frames...

**Comment by Lit-fish-auto** [comment]
@andlaus
cantools generate c resource  can be C++ used?
above all , thank a lot.

**Comment by andlaus** [comment]
It generates C code, but C is a subset of C++. So yes, it generates C++ code.

**Comment by andlaus** [comment]
you have to use the `cantools` command on the command line. That said, if you're not already at least somewhat proficient in C/C++, I would advise you not to waste your time and hire some consultant to do the job for you...
